3rdparty/mshadow/mshadow/base.h

```diff
@@ -839,7 +840,7 @@ MSHADOW_XINLINE bool MaxValue<bool>(void) {
 /*! \brief maximum value of uint32_t */
 template<>
 MSHADOW_XINLINE uint32_t MaxValue<uint32_t>(void) {
-  return -1;
+  return std::numeric_limits<uint32_t>::max();
```

Comment:
What's the indication of this change? Would it break some existing logic?

Reply:
Fixes a compiler warning treated as error due to `-WError` on CI.
Essentially we're currently relying on unsigned integer overflow to obtain the max value. `std::numeric_limits<uint32_t>::max();` gives the max value directly

Reply:
Thanks. I am actually not questioning this change which I think is correct. I am wondering if you have checked places where the return value of "-1" from this function is expected and if this will break that assumption.